synapse/handlers/directory.py

```diff
@@ -174,6 +174,8 @@ async def delete_association(
                 is neither the creator of the alias, nor a server admin.
 
             SynapseError: if the alias belongs to an AS
+
+            ShadowBanError: if the requester is shadow-banned.
```

Comment:
is this coming from `_update_canonical_alias`? should maybe just ignore the exception if so, as we do with AuthError.

Reply:
also: would be nice to stick the `Raises` docstring on  `_update_canonical_alias`, even though it's internal.